index.bs

```diff
@@ -894,6 +933,52 @@ To <dfn>parse event triggers</dfn> given an [=ordered map=] |map|:
 Issue: Allow the user agent to return null if |values|'s [=list/size=] is
 greater than some constant.
 
+To <dfn>parse aggregatable trigger data</dfn> given an [=ordered map=] |map|:
+
+1. Let |aggregatableTriggerData| be a new [=list=].
+1. If |map|["`aggregatable_trigger_data`"] does not [=map/exist=], return |aggregatableTriggerData|.
+1. Let |values| be |map|["`aggregatable_trigger_data`"].
+1. If |values| is not a [=list=], return null.
+1. If |values|'s [=list/size=] is greater than the user agent's
+    [=max aggregatable trigger data per trigger=], return null.
+1. [=list/iterate|For each=] |value| of |values|:
+    1. If |value| is not an [=ordered map=], return null.
+    1. If |value|["`key_piece`"] does not [=map/exist=] or is not a [=string=], return null.
+    1. Let |keyPiece| be the result of running [=parse an aggregation key piece=] with |value|["`key_piece`"].
+    1. If |keyPiece| is an error, return null.
+    1. If |value|["`source_keys`"] does not [=map/exist=] or is not a [=list=], return null.
+    1. If |value|["`source_keys`"]'s [=list/size=] is greater than the user agent's
+        [=max aggregation keys per attribution=], return null.
+    1. Let |sourceKeys| be a new [=set=].
+    1. [=list/iterate|For each=] |sourceKey| of |value|["`source_keys`"]:
+        1. If |sourceKey| is not a [=string=], return null.
+        1. [=set/Append=] |sourceKey| to |sourceKeys|.
+    1. Let |aggregatableTrigger| be a new [=aggregatable trigger data=] with the items:
+        : [=aggregatable trigger data/key piece=]
+        :: |keyPiece|
```

Comment:
I think it could be the same, like applying the same `key_piece` to different `source_keys` depending on the filters. But I'm not sure whether there's such use case in reality.